src/store/models/network.ts

```diff
@@ -922,6 +935,95 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  autoBalanceChannels: thunk(async (actions, { id }, { getState, getStoreState }) => {
+    const { networks } = getState();
+    const network = networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    const getNodeLightningService = (node: LightningNode): LightningService => {
+      switch (node.implementation) {
+        case 'LND':
+          return lndService;
+        case 'c-lightning':
+          return clightningService;
+        case 'eclair':
+          return eclairService;
+        default:
+          throw new Error('unknown implementation');
+      }
+    };
+
+    const balanceChannel = async (
+      channel: LightningNodeChannel,
+      localNode: LightningNode,
+      remoteNode: LightningNode,
+      satsTolerance = 150,
+    ) => {
+      if (channel.status !== 'Open') {
+        // TODO: warn about channel not opened.
+        return;
+      }
+
+      if (remoteNode === undefined || remoteNode === null) {
+        // TODO: warn about remote node being null.
+        return;
+      }
+
+      const localBalance = Number(channel.localBalance);
+      const remoteBalance = Number(channel.remoteBalance);
+      const toPay = Math.floor(Math.abs(localBalance - remoteBalance) / 2);
+
+      // If the balance difference in satoshis is too small, we ignore it.
+      if (toPay < satsTolerance) {
+        return;
+      }
+
+      // The source node pays an invoice to the target node, in order to balance the channel.
+      const src = localBalance > remoteBalance ? localNode : remoteNode;
+      const target = localBalance > remoteBalance ? remoteNode : localNode;
+
+      console.log(
+        '[AUTO BALANCE]: ',
+        'paying from ' + src.name + ' to ' + target.name + ' the amount ' + toPay,
+      );
+
+      const invoice = await getNodeLightningService(target).createInvoice(target, toPay);
+
+      await getNodeLightningService(src).payInvoice(src, invoice);
+    };
+
+    interface ChannelInfo {
```

Comment:
It's not really necessary to declare an invoice inside of a function. The purpose of interfaces is to be reusable throughout the codebase. Since it's going to only be used once here in this function, declare the structure inline.
```ts
const channelsInfo: {
  channel: LightningNodeChannel;
  fromNode: LightningNode;
  toNode: LightningNode;
}[] = [];
```